tests/functionnal/CommonDBTM.php

```diff
@@ -329,4 +329,446 @@ function () use ($DB) {
+      $this->boolean($rel->canEdit($idr[7]))->isTrue();
+
+      $input = [
+         'contacts_id' =>  $idc[2], // Child 1
+         'suppliers_id' => $ids[3]  // Child 2
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isFalse();
+
+      // See only in child entity 2 + parent if recursive
+      $this->boolean(\Session::changeActiveEntities($ent2))->isTrue();
+
+      $this->boolean($rel->can($idr[0], READ))->isFalse();  // root / root
+      $this->boolean($rel->canEdit($idr[0]))->isFalse();
+      $this->boolean($rel->can($idr[1], READ))->isFalse();  // root / root rec
+      $this->boolean($rel->canEdit($idr[1]))->isFalse();
+      $this->boolean($rel->can($idr[2], READ))->isFalse();  // root rec / root
+      $this->boolean($rel->canEdit($idr[2]))->isFalse();
+      $this->boolean($rel->can($idr[3], READ))->isTrue();   // root rec / root rec
+      $this->boolean($rel->canEdit($idr[3]))->isFalse();
+      $this->boolean($rel->can($idr[4], READ))->isFalse();  // root rec / child 1
+      $this->boolean($rel->canEdit($idr[4]))->isFalse();
+      $this->boolean($rel->can($idr[5], READ))->isTrue();   // root rec / child 2
+      $this->boolean($rel->canEdit($idr[5]))->isTrue();
+      $this->boolean($rel->can($idr[6], READ))->isFalse();  // child 1 / root rec
+      $this->boolean($rel->canEdit($idr[6]))->isFalse();
+      $this->boolean($rel->can($idr[7], READ))->isFalse();  // child 1 / child 1
+      $this->boolean($rel->canEdit($idr[7]))->isFalse();
+
+      $input = [
+         'contacts_id' =>  $idc[0], // root
+         'suppliers_id' => $ids[0]  // root
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isFalse();
+
+      $input = [
+         'contacts_id'  =>  $idc[0],// root
+         'suppliers_id' => $ids[1]  // root + rec
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isFalse();
+
+      $input = [
+         'contacts_id'  =>  $idc[1],// root + rec
+         'suppliers_id' => $ids[0]  // root
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isFalse();
+
+      $input = [
+         'contacts_id' =>  $idc[3], // Child 2
+         'suppliers_id' => $ids[0]  // root
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isFalse();
+
+      $input = [
+         'contacts_id'  =>  $idc[3],// Child 2
+         'suppliers_id' => $ids[1]  // root + rec
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isTrue();
+      $idr[7] = (int)$rel->add($input);
+      $this->integer($idr[7])->isGreaterThan(0);
+      $this->boolean($rel->can($idr[7], READ))->isTrue();
+      $this->boolean($rel->canEdit($idr[7]))->isTrue();
+
+      $input = [
+         'contacts_id' =>  $idc[3], // Child 2
+         'suppliers_id' => $ids[2]  // Child 1
+      ];
+      $this->boolean($rel->can(-1, CREATE, $input))->isFalse();
+
+      $input = [
+         'contacts_id' =>  $idc[3], // Child 2
+         'suppliers_id' => $ids[3]  // Child 3
```

Comment:
Comment is wrong : should be `$ids[3]  // Child 2` instead of `$ids[3]  // Child 3`.